src/machine.rs

```diff
@@ -986,7 +986,7 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
         dest: &MPlaceTy<'tcx, Provenance>,
         ret: Option<mir::BasicBlock>,
         unwind: mir::UnwindAction,
-    ) -> InterpResult<'tcx, Option<ty::Instance<'tcx>>> {
+    ) -> InterpResult<'tcx> {
```

Comment:
Not sure what this is about, but this can't be right. Are you maybe using an outdated local toolchain? I recommend doing `touch .auto-toolchain` in your Miri checkout so that all `./miri` command will automatically install the correct toolchain when it is outdated.

Reply:
I see was not sure, why I kept in separate commit. Will fix later.

Reply:
If you are having trouble running Miri on your own system, feel free to ask [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/269128-miri); there's people there that can help. You shouldn't have to always wait for CI to see if anything worked. :)

You don't even need a Solaris machine; `./miri run -v --target x86_64-pc-solaris tests/pass/vec.rs` should do it. (This is assuming you already did `touch .auto-toolchain`.)